library/src/main/java/com/schibsted/spain/barista/BaristaClickActions.java

```diff
@@ -15,17 +18,43 @@
 
 public class BaristaClickActions {
 
-  public static void click(@IdRes int id) {
+  private static final ResourceTypeChecker RESOURCE_TYPE_CHECKER = new ResourceTypeChecker();
+
+  public static void click(int id) {
+    if (isStringResource(id)) {
+      clickByString(id);
+    } else if (isIdResource(id)) {
+      clickById(id);
+    } else {
+      throw new BaristaArgumentTypeException();
+    }
+  }
+
+  private static void clickById(@IdRes int id) {
+    try {
+      scrollAndClickViewById(id);
+    } catch (AmbiguousViewMatcherException multipleViewsMatched) {
+      try {
+        scrollAndClickDisplayedViewById(id);
+      } catch (PerformException parentIsNotAnScrollView) {
+        clickDisplayedViewById(id);
+      }
+    } catch (PerformException parentIsNotAnScrollView) {
+      clickDisplayedViewById(id);
+    }
+  }
+
+  private static void clickByString(@StringRes int id) {
     try {
-      scrollAndClickView(id);
+      scrollAndClickViewByString(id);
     } catch (AmbiguousViewMatcherException multipleViewsMatched) {
       try {
-        scrollAndClickDisplayedView(id);
+        scrollAndClickDisplayedViewByString(id);
       } catch (PerformException parentIsNotAnScrollView) {
-        clickDisplayedView(id);
+        clickDisplayedViewByString(id);
       }
     } catch (PerformException parentIsNotAnScrollView) {
-      clickDisplayedView(id);
+      clickDisplayedViewByString(id);
```

Comment:
Hmm with this change each click method is not triplicated.
I think it's better to extract a method that receives a matcher, and pass the matcher from the generic method (`withText("")`, `withText(res)`, `withId(id)`).

Did I explained myself good enough? What do you think?